addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4560,6 +4560,15 @@ var SnippetsMenu = Widget.extend({
                 }
             }
 
+            const invisibleSnippetSection = this.getEditableArea()
+                .find('.o_snippet_mobile_invisible, .o_snippet_desktop_invisible')
+                .toArray()
+                .find((snippetSection) => {
+                    return snippetSection.dataset.invisible === '1';
+                });
+            if (invisibleSnippetSection) {
+                this._activateSnippet(false);
+            }
```

Comment:
What you do here is: if an element is invisible on the page, it deactivates the snippet.
This is wrong, your change introduces a worse bug 😬 . It deactivates any selected snippet (even not invisible one) when switching to desktop view if there is an invisible element on the page.
See video:

https://github.com/user-attachments/assets/21d084d5-88fd-444b-9b99-1d584b440682

Also I don't think you need to find again invisible elements, it's already done a few lines above (line 4553)

Reply:
@bvr-odoo I have pushed changes! It meets expected behaviour shown in video. I have used getSelection method to recognize currently selected element and compares it with all the invisible element present in DOM then refresh options accordingly.
Thank you!